### PR TITLE
Propose some contribution guidelines.

### PR DESCRIPTION
**Pull request by andrewgordstewart**
This
- copies the PR template from our statechannels mono repo
- defines some contribution guidelines

I feel we want to update our PR template. This draft PR enables discussion on what should be kept/added/removed.

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+
+Before spending lots of time working on an issue, consider asking us for feedback. We would love to help make your contributions more successful!
+
+Pull requests (internal or external) will be reviewed by one or two members of Magmo. We aim to review most contributions within one business day.
+
+We include a PR template to serve as a **guideline**, not a **rule**, to communicate the code culture we wish to maintain in this repository.
+
+## Style
+
+As of now, Magmo is new to the go language, and has no specific opinions about go style. As a rule of thumb, we defer to the [Effective Go](https://go.dev/doc/effective_go) document [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences) as "style guides".
```
Comment (andrewgordstewart) [inline contributing.md:19]:
```suggestion
As of now, Magmo is new to the go language, and has no specific opinions about go style. As a rule of thumb, we defer to the [Effective Go](https://go.dev/doc/effective_go) document [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) as "style guides".
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+- [ ] I have commented my code wherever necessary (can be 0)
+- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
+
+### Project management
+
+- [ ] I have applied the [appropriate labels](https://www.notion.so/statechannels/Team-working-agreements-3cbcd6e85a7e481db4fe572ddf50cbd6#f959a79c3b4f41708b8506612a99ec14)
+- [ ] I have [linked to all relevant issues](https://help.zenhub.com/support/solutions/articles/43000010350-connecting-pull-requests-to-github-issues) (can be 0)
+- [ ] I have [linked to all dependent issues](https://help.zenhub.com/support/solutions/articles/43000010349-create-github-issue-dependencies) (can be 0)
+- [ ] I have assigned myself to this PR
+- [ ] I have chosen the appropriate [pipeline](https://www.notion.so/statechannels/Team-working-agreements-3cbcd6e85a7e481db4fe572ddf50cbd6#a96b6b02704d46afbcff147cf5a85566) on zenhub for the linked issue
```
Comment (geoknee) [inline .github/pull_request_template.md:58]: We should remove references to zenhub. And probably also to notion, unless we are opening that up to the public.

**Review by geoknee — APPROVED** [review]
I think this looks pretty good. If we merge soon, I think we will get good feedback from folks actually using the template and can update if necessary.

File: contributing.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+
+Contributions to go-nitro are welcome from anyone.
+
+Magmo (at Consensys Mesh) are developing go-nitro under contract for the Filecoin Foundation. As a consequence, we are obligated to maintain a certain level of quality.
+
+## Contribution lifecycle
+
+Magmo uses [Github Projects](https://github.com/statechannels/go-nitro/projects/1) for go-nitro project management. We will aim to keep a handful of "good first issues" in [the pipeline](https://github.com/statechannels/go-nitro/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22), which are a good place to start.
+
+Before spending lots of time working on an issue, consider asking us for feedback. We would love to help make your contributions more successful!
```
Comment (NiloCK) [inline contributing.md:11]: Consider:
```suggestion
Before spending lots of time working on an issue, consider asking us for feedback via [Slack](https://statechannels.slack.com/archives/C02J81JFD3J), the [issue tracker](https://github.com/statechannels/go-nitro/issues), or [email](magmo@mesh.xyz). We would love to help make your contributions more successful!
```

caveat: I'm not sure whether that email link will render as a `mailto`.

File: contributing.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
+# Contributing
+
+Contributions to go-nitro are welcome from anyone.
+
+Magmo (at Consensys Mesh) are developing go-nitro under contract for the Filecoin Foundation. As a consequence, we are obligated to maintain a certain level of quality.
```
Comment (NiloCK) [inline contributing.md:5]:
```suggestion
Magmo (at Consensys Mesh) are developing go-nitro under contract for the Filecoin Foundation. As a consequence, we reserve discretionary veto rights on feature additions, architectural decisions, etc., which may hinder progress toward the specific aims of this contract.
```

**Comment by NiloCK** [comment]
off topic nit: 

`configuredRemote/branchName` is a convention in git, so branch names like `ags/feature` may be better named as `ags.feature`, `ags-feature`, etc - just use some other delimiter.

**Comment by NiloCK** [comment]
on topic:

agreed with George. This is a good time for this to go in, and it's good!

File: .github/pull_request_template.md (hunk 2 above)
Comment (lalexgap) [reply .github/pull_request_template.md]: I've removed most of this in 6fd4e32

File: contributing.md [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+
+Pull requests (internal or external) will be reviewed by one or two members of Magmo. We aim to review most contributions within one business day.
+
+We include a PR template to serve as a **guideline**, not a **rule**, to communicate the code culture we wish to maintain in this repository.
+
+## Style
+
+When in doubt, defer to the [Effective Go](https://go.dev/doc/effective_go) document [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) as "style guides".
+
+## Judgement
```
[the remark this answers is not in the document]
Comment (lalexgap) [reply contributing.md]: Done in a229754

**Pull request merged**